public/app/core/logs_model.ts

```diff
@@ -140,52 +135,29 @@ export function makeSeriesForLogs(sortedRows: LogRowModel[], bucketSize: number,
   return seriesList.map((series, i) => {
     series.datapoints.sort((a: number[], b: number[]) => a[1] - b[1]);
 
-    // EEEP: converts GraphSeriesXY to DataFrame and back again!
     const data = toDataFrame(series);
     const fieldCache = new FieldCache(data);
 
-    const timeField = fieldCache.getFirstFieldOfType(FieldType.time)!;
-    timeField.display = getDisplayProcessor({
-      field: timeField,
-      timeZone,
-      theme: config.theme2,
-    });
-
     const valueField = fieldCache.getFirstFieldOfType(FieldType.number)!;
-    valueField.config = {
-      ...valueField.config,
-      color: series.color,
-    };
-
-    valueField.name = series.alias;
-    const fieldDisplayProcessor = getDisplayProcessor({ field: valueField, timeZone, theme: config.theme2 });
-    valueField.display = (value: any) => ({ ...fieldDisplayProcessor(value), color: series.color });
-
-    const points = getFlotPairs({
-      xField: timeField,
-      yField: valueField,
-      nullValueMode: NullValueMode.Null,
-    });
 
-    const graphSeries: GraphSeriesXY = {
-      color: series.color,
-      label: series.alias,
-      data: points,
-      isVisible: true,
-      yAxis: {
-        index: 1,
-        min: 0,
-        tickDecimals: 0,
+    data.fields[valueField.index].config.min = 0;
+    data.fields[valueField.index].config.decimals = 0;
+
+    data.fields[valueField.index].config.custom = {
+      pathBuilder: uPlot.paths.bars!({ align: 0, size: [0.9, 5] }),
```

Comment:
maybe expose something more simple?

we already have `custom.barAlignment`, so potentially add `custom.barWidthFactor` and `custom.barMaxWidth`.

in the future we might expose control over these in the TimeSeries panel anyhow, so this will have us covered for that case, too :)

Reply:
Makes sense 👍 One question, though. uPlot uses single [size property containing both - width factor and max width](https://github.com/leeoniya/uPlot/blob/69fb492b7f0069c5ef910f0cec5091964f2ccd78/src/paths/bars.js#L7). `custom.barWidthFactor` to 0.6 and `custom.barMaxWidth` should be optional so I can default `custom.barWidthFactor` to 0.6 and `custom.barMaxWidth` to `Infinity` as in uPlot but it might be confusing if uPlot defaults changed in the future.

Would it make sense to default to current uPlot defaults ([0.6, Infinity]) for now, but change uPlot so both values can be provided separately?

Reply:
i dont think we should rely on uPlot's defaults at all. if we want to make each optional in the fieldConfig and simply set the default we want in UPlotSeriesBuilder.ts, that should work fine.

right now they're hard-coded, but we can read them from args with defaults.

https://github.com/grafana/grafana/blob/888683a8ad2f0f05d66d5a5d085c230dbba26e33/packages/grafana-ui/src/components/uPlot/config/UPlotSeriesBuilder.ts#L173-L174

would that work?

Reply:
err, sorry. i think maybe for now it's best to pre-initialize a `barsExplore` pathbuilder variant with hardcoded settings and choose in SeriesBuilder based on matching the same hardcoded values of barWidthFactor and barMaxWidth.

that will keep this PR from turning into a bigger refactor. we can do a followup at some point and make this less hacky.

Reply:
I think it should require too many changes. How about this? https://github.com/grafana/grafana/pull/34531/commits/f8e44b01c7efb4192af67d1a37cfd276adfa1e32